catalog issue does not give good user/error info

reported from @phillip-kruger / @myfear
```
jbang quarkus-agent-mcp@quarkusio
[jbang] [ERROR] Unknown catalog 'quarkusio'
[jbang] Run with --verbose or -x for more details. The --verbose or -x must be placed before the jbang command. I.e. jbang --verbose run [...]
```

verbose says:

```
jbang --verbose catalog list quarkusio
[jbang] [0:139] jbang version 0.138.0
[jbang] [0:145] Reading catalog from /Users/meisele/.jbang/implicit-catalog.json
[jbang] [0:159] Reading catalog from classpath:/jbang-catalog.json
[jbang] [0:212] Requesting HTTP GET https://raw.githubusercontent.com/jbanghub/jbang-catalog/main/jbang-catalog.json
[jbang] [0:212] Headers {If-None-Match=["a2e9c25832a706272cfb8dc07eaea748d414c23e03670932fc3b9be3eb3a98f9"], User-Agent=[JBang/0.138.0 (Mac OS X/26.5.1/aarch64) avaJ/25/Eclipse Adoptium], If-Modified-Since=[Wed, 24 Jun 2026 09:29:59 GMT]}
[jbang] [0:572] Unable to read catalog https://raw.githubusercontent.com/jbanghub/jbang-catalog/main/jbang-catalog.json because dev.jbang.resources.ResourceNotFoundException: Could not download https://raw.githubusercontent.com/jbanghub/jbang-catalog/main/jbang-catalog.json
[jbang] [0:572] Local catalog 'quarkusio' not found, trying implicit catalogs...
[jbang] [0:576] Requesting HTTP GET https://raw.githubusercontent.com/quarkusio/jbang-catalog/HEAD/jbang-catalog.json
[jbang] [0:576] Headers {If-None-Match=["09b599fa765d3c0b6905752f91095922295d53c135587bee25eb73bb3cd20471"], User-Agent=[JBang/0.138.0 (Mac OS X/26.5.1/aarch64) avaJ/25/Eclipse Adoptium], If-Modified-Since=[Wed, 24 Jun 2026 09:29:59 GMT]}
[jbang] [0:695] No catalog found at https://github.com/quarkusio/jbang-catalog/blob/HEAD/jbang-catalog.json
dev.jbang.resources.ResourceNotFoundException: Could not download https://github.com/quarkusio/jbang-catalog/blob/HEAD/jbang-catalog.json
```

user could use browser and curl just fine to get the catalog url and no known vpn/proxy configured.

turns out issue is GITHUB_TOKEN was set to out of  date token. when unsetting GITHUB_TOKEN all was good.

Things we should look into improving:

1) original message don't actually explain the catalog is Unknown because it is not possible to download...that would be first hint
2) --verbose dont actually log the exceptions...we should add that
3) no hint about GITHUB_TOKEN being used in the download....we should probably verbose log that too

--- a/src/main/java/dev/jbang/catalog/Catalog.java
+++ b/src/main/java/dev/jbang/catalog/Catalog.java
@@ -327,7 +327,7 @@ public class Catalog {
 					if (result != null)
 						return result;
 				} catch (Exception e) {
-					Util.verboseMsg("Unable to read catalog " + cr.catalogRef + " because " + e);
+					Util.verboseMsg("Unable to read catalog " + cr.catalogRef + ": " + e.getMessage(), e);
 				}
 			}
 		}

--- a/src/main/java/dev/jbang/catalog/ImplicitCatalogRef.java
+++ b/src/main/java/dev/jbang/catalog/ImplicitCatalogRef.java
@@ -7,6 +7,7 @@ import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Stream;
 
+import dev.jbang.util.NetUtil;
 import dev.jbang.util.Util;
 
 public class ImplicitCatalogRef {
@@ -88,6 +89,12 @@ public class ImplicitCatalogRef {
 			return Optional.of(url);
 		} catch (Exception ex) {
 			Util.verboseMsg("No catalog found at " + url, ex);
+			String authMethod = NetUtil.describeAuthMethod(url);
+			if (authMethod != null) {
+				Util.warnMsg("Could not download catalog from " + url
+						+ ". Authentication via " + authMethod + " was used but may be expired or invalid."
+						+ " Try updating or removing the credentials.");
+			}
 			return Optional.empty();
 		}
 	}

--- a/src/main/java/dev/jbang/util/NetUtil.java
+++ b/src/main/java/dev/jbang/util/NetUtil.java
@@ -641,14 +641,60 @@ public class NetUtil {
 	}
 
 	private static boolean isAGithubUrl(URLConnection urlConnection) {
-		String host = urlConnection.getURL().getHost();
+		return isAGithubHost(urlConnection.getURL().getHost());
+	}
+
+	private static boolean isAGithubHost(String host) {
 		return host.equals("github.com") || host.endsWith(".github.com")
 				|| host.equals("githubusercontent.com") || host.endsWith(".githubusercontent.com");
 	}
 
 	private static boolean isAGitlabUrl(URLConnection urlConnection) {
-		String host = urlConnection.getURL().getHost();
+		return isAGitlabHost(urlConnection.getURL().getHost());
+	}
+
+	private static boolean isAGitlabHost(String host) {
 		return host.equals("gitlab.com") || host.endsWith(".gitlab.com");
+	}
+
+	/**
+	 * Returns a human-readable description of the authentication method that would
+	 * be used for the given URL, or {@code null} if no authentication is
+	 * configured.
+	 */
+	public static String describeAuthMethod(String url) {
+		try {
+			URL u = new URL(url);
+			String host = u.getHost();
+
+			// 1. Well-known env vars for specific hosts
+			if (isAGithubHost(host) && !isNullOrBlankString(System.getenv("GITHUB_TOKEN"))) {
+				return "GITHUB_TOKEN";
+			}
+			if (isAGitlabHost(host) && !isNullOrBlankString(System.getenv("GITLAB_TOKEN"))) {
+				return "GITLAB_TOKEN";
+			}
+
+			// 2. URL userinfo
+			if (u.getUserInfo() != null) {
+				return "URL credentials";
+			}
+
+			// 3. .netrc
+			NetrcParser.NetrcEntry entry = getNetrc().getEntry(host).orElse(null);
+			if (entry != null && !isNullOrBlankString(entry.getLogin()) && !isNullOrBlankString(entry.getPassword())) {
+				return ".netrc";
+			}
+
+			// 4. Global basic auth env vars
+			if (!isNullOrBlankString(System.getenv(JBANG_AUTH_BASIC_USERNAME))
+					&& !isNullOrBlankString(System.getenv(JBANG_AUTH_BASIC_PASSWORD))) {
+				return "JBANG_AUTH_BASIC_USERNAME/PASSWORD";
+			}
+		} catch (Exception e) {
+			// ignore
+		}
+		return null;
 	}
 
 	public static String getDispositionFilename(String disposition) {

--- a/src/test/java/dev/jbang/util/TestDescribeAuthMethod.java
+++ b/src/test/java/dev/jbang/util/TestDescribeAuthMethod.java
@@ -1,0 +1,85 @@
+package dev.jbang.util;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.nullValue;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import dev.jbang.BaseTest;
+
+/**
+ * Tests for {@link NetUtil#describeAuthMethod(String)}.
+ */
+public class TestDescribeAuthMethod extends BaseTest {
+
+	@TempDir
+	Path tempDir;
+
+	@AfterEach
+	void cleanup() {
+		NetUtil.resetNetrcCache();
+	}
+
+	private void setNetrc(String content) throws IOException {
+		Path netrc = tempDir.resolve(".netrc");
+		Files.writeString(netrc, content);
+		NetUtil.setNetrcFile(netrc);
+	}
+
+	@Test
+	void testUrlCredentialsDetected() {
+		assertThat(NetUtil.describeAuthMethod("https://user:pass@example.com/file.java"),
+				equalTo("URL credentials"));
+		// Non-matching URL without credentials returns null
+		assertThat(NetUtil.describeAuthMethod("https://example.com/file.java"), nullValue());
+	}
+
+	@Test
+	void testNetrcDetected() throws IOException {
+		setNetrc("machine myserver.example.com\nlogin admin\npassword secret\n");
+		assertThat(NetUtil.describeAuthMethod("https://myserver.example.com/path/file.java"),
+				equalTo(".netrc"));
+		// Host not in .netrc returns null
+		assertThat(NetUtil.describeAuthMethod("https://otherserver.example.com/file.java"), nullValue());
+	}
+
+	@Test
+	void testNetrcDefaultEntryDetected() throws IOException {
+		setNetrc("default\nlogin fallback\npassword fallbackpass\n");
+		assertThat(NetUtil.describeAuthMethod("https://any-host.example.com/file.java"),
+				equalTo(".netrc"));
+	}
+
+	@Test
+	void testNoAuthReturnsNull() throws IOException {
+		// Set an empty .netrc so it won't match anything
+		setNetrc("");
+		assertThat(NetUtil.describeAuthMethod("https://example.com/file.java"), nullValue());
+	}
+
+	@Test
+	void testNetrcNoMatchFallsThrough() throws IOException {
+		setNetrc("machine other.example.com\nlogin user\npassword pass\n");
+		assertThat(NetUtil.describeAuthMethod("https://nomatch.example.com/file.java"), nullValue());
+	}
+
+	@Test
+	void testInvalidUrlReturnsNull() {
+		assertThat(NetUtil.describeAuthMethod("not-a-url"), nullValue());
+	}
+
+	@Test
+	void testUrlCredentialsTakePriorityOverNetrc() throws IOException {
+		setNetrc("machine example.com\nlogin netrcuser\npassword netrcpass\n");
+		// URL credentials (step 2) should win over .netrc (step 3)
+		assertThat(NetUtil.describeAuthMethod("https://user:pass@example.com/file.java"),
+				equalTo("URL credentials"));
+	}
+}

